browser/skus/sdk_controller_factory.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_SKUS_SDK_CONTROLLER_FACTORY_H_
+#define BRAVE_BROWSER_SKUS_SDK_CONTROLLER_FACTORY_H_
+
+#include "base/memory/singleton.h"
+#include "brave/components/skus/browser/sdk_controller.h"
+#include "components/keyed_service/content/browser_context_keyed_service_factory.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+
+namespace skus {
+
+// Singleton that creates/deletes SdkControllerFactory as new Profiles are
+// created/shutdown.
+class SdkControllerFactory : public BrowserContextKeyedServiceFactory {
+ public:
+  static mojo::PendingRemote<mojom::SdkController> GetForContext(
+      content::BrowserContext* context);
+  static skus::SdkController* GetControllerForContext(
```

Comment:
please remove this method, it should be private in wallet only for internal classes

Reply:
see https://github.com/brave/brave-core/pull/11121/files#r775662245 except that in this case there should not even be a private method that exposes the raw pointer.

Reply:
Addressed in https://github.com/brave/brave-core/pull/10847/commits/da6c3e0611ba82762e44480b78e18afc06f2fc29 - let me know what you think